tests/FakeItEasy.Specs/ConfigurationSpecs.cs

```diff
@@ -580,11 +602,48 @@ public static void CallsBaseMethodAndDoesNothing(BaseClass fake)
             "And I configure all methods to invoke an action"
                 .x(() => A.CallTo(fake).Invokes(() => { }));
 
-            "When I call a reference type method"
+            "When I call a non-fakeable reference type method"
                 .x(() => result = fake.Bas());
 
-            "Then it returns null"
-                .x(() => result.Should().BeNull());
+            "Then it returns the same value as an unconfigured fake"
+                .x(() => result.Should().Be(A.Fake<IFoo>().Bas()));
+        }
+
+        [Scenario]
+        public static void InvokesAfterStrictFakeableableReferenceTypeKeepsDefaultReturnValue(
+            IFoo fake,
+            IFoo result)
+        {
+            "Given a strict fake"
+                .x(() => fake = A.Fake<IFoo>(options => options.Strict()));
+
+            "And I configure all methods to invoke an action"
+                .x(() => A.CallTo(fake).Invokes(() => { }));
+
+            "When I call a fakeable reference type method"
+                .x(() => result = fake.Bafoo());
+
+            "Then it returns the same value as an unconfigured fake"
+                .x(() => result.Should().Be(A.Fake<IFoo>().Bafoo()));
+        }
+
+        [Scenario]
+        public static void AssignsOutAndRefParametersForAllMethodsKeepsDefaultReturnValue(

```

Comment:
Didn't require a change to make it pass, but I wanted to be explicit about the requirement.